Code/Mantid/Framework/WorkflowAlgorithms/src/ConvolutionFitSequential.cpp

```diff
@@ -0,0 +1,650 @@
+/** Execute the algorithm.
+ */
+void ConvolutionFitSequential::exec() {
+  // Initialise variables with properties
+  MatrixWorkspace_sptr inputWs = getProperty("InputWorkspace");
+  const std::string function = getProperty("Function");
+  const std::string backType =
+      convertBackToShort(getProperty("backgroundType"));
+  const double startX = getProperty("StartX");
+  const double endX = getProperty("EndX");
+  const int specMin = getProperty("SpecMin");
+  const int specMax = getProperty("Specmax");
+  const bool convolve = getProperty("Convolve");
+  const int maxIter = getProperty("MaxIterations");
+  const std::string minimizer = getProperty("Minimizer");
+
+  // Inspect function to obtain fit Type and background
+  std::vector<std::string> functionValues = findValuesFromFunction(function);
+  const std::string LorentzNum = functionValues[0];
+  const std::string funcName = functionValues[1];
+  bool usingLorentzians = false;
+  if (boost::lexical_cast<int>(LorentzNum) > 0) {
+    usingLorentzians = true;
+  }
+
+  // Check if a delta function is being used
+  bool delta = false;
+  std::string usingDelta = "false";
+  auto pos = function.find("Delta");
+  if (pos != std::string::npos) {
+    delta = true;
+    usingDelta = "true";
+  }
+
+  // Add logger information
+  m_log.information("Input files: " + inputWs->getName());
+  m_log.information("Fit type: Delta=" + usingDelta + "; Lorentzians=" +
+                    LorentzNum);
+  m_log.information("Background type: " + backType);
+
+  // Output workspace name
+  std::string outputWsName = inputWs->getName();
+  pos = outputWsName.rfind("_");
+  if (pos != std::string::npos) {
+    outputWsName = outputWsName.substr(0, pos + 1);
+  }
+  outputWsName += "conv_";
+  if (delta) {
+    outputWsName += "Delta";
+  }
+  if (LorentzNum.compare("0") != 0) {
+    outputWsName += LorentzNum + "L";
+  } else {
+    outputWsName += convertFuncToShort(funcName);
+  }
+  outputWsName += backType + "_s";
+  outputWsName += boost::lexical_cast<std::string>(specMin);
+  outputWsName += "_to_";
+  outputWsName += boost::lexical_cast<std::string>(specMax);
+
+  // Convert input workspace to get Q axis
+  const std::string tempFitWsName = "__convfit_fit_ws";
+  auto tempFitWs = convertInputToElasticQ(inputWs, tempFitWsName);
+
+  // Fit all spectra in workspace
+  std::string plotPeakInput = "";
+  for (int i = 0; i < specMax + 1; i++) {
+    std::string nextWs = tempFitWsName + ",i";
+    nextWs += boost::lexical_cast<std::string>(i);
+    plotPeakInput += nextWs + ";";
+  }
+
+  // passWSIndex
+  bool passIndex = false;
+  if (funcName.find("Diff") != std::string::npos ||
+      funcName.find("Stretched") != std::string::npos) {
+    passIndex = true;
+  }
+
+  // Run PlotPeaksByLogValue
+  auto plotPeaks = createChildAlgorithm("PlotPeakByLogValue", 0.0, 0.70, true);
+  plotPeaks->setAlwaysStoreInADS(true);
```

Comment:
Is there a reason that the algorithms are storing the intermediate workspaces in the ADS? It would be better not to do this as then they won't see the light of day unless explicitly required.